test/ttmlir/Dialect/TTNN/optimizer/bf_interleaved_policy/fork_join_02.mlir

```diff
@@ -0,0 +1,35 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="enable-optimizer=true memory-layout-analysis-enabled=true memory-layout-analysis-policy=BFInterleaved" %s | FileCheck %s
+//
+//         A
+//         |
+//         B
+//       /   \
+//      C     D
+//       \   /
+//         E
+//         |
+//         F
+//
+// There is enough L1 memory to schedule this fork-join without any DRAM spill even though
+// output tensors of C and D cannot be in L1 memory at the same time.
+//
+module attributes {} {
+  func.func @forward(%arg0: tensor<4096x5120xbf16>, %arg1: tensor<5120x1024xbf16>, %arg2: tensor<5120x1024xbf16>) -> tensor<4096x1024xbf16> {
```

Comment:
Also add another test where not everything can fit into L1. E.g. the output of C is too big, everything else can fit into L1.